### ISSUE TITLE
Promise' is not defined. (no-undef)

### ISSUE THREAD

**Issue by bharathvaj1995** [eslint/eslint#9812]
* **ESLint Version:** 4.14.0
* **Node Version:** 6.11.1
* **npm Version:** 5.3.0

**What parser (default, Babel-ESLint, etc.) are you using?**
default

```js
const promise = new Promise((resolve, reject) => {});
```
**eslint config**
```js
{
	"env": {
		"browser": true,
		"node": true
	},
	"parserOptions": {
		"ecmaVersion": 6,
		"sourceType": "script"
	},
	"extends": ["kentcdodds/possible-errors", "kentcdodds/best-practices"],
	"rules": {
		"global-require": "off"
	}
}
```
**What did you expect to happen?**
No un-def error for es6 globals (**Promise**)

**What actually happened?**
Throws **un-def error** for 'Promise'

**Comment by j-f1** [comment]
Do you have `es6: true` set under the `env` key in your ESLint config?

**Comment by bharathvaj1995** [comment]
@j-f1 I have updated my .eslintrc file

**Comment by j-f1** [comment]
@bharathvaj1995 It looks like you only have `browser` and `node` set in your `env` setting. Can you try adding `"es6": true` there, too?

**Comment by platinumazure** [comment]
@bharathvaj1995 ecmaVersion only affects the parser, so it allows ES6 *syntax* (such as let/const). The environment will set ecmaVersion for you, but will also enable all of the ES6 *globals* (such as Promise, Map, Set). If you want ES7 (or later) and ES6 globals, you would do env es6:true and ecmaVersion: (something greater than 6)... Hope that makes sense!

**Comment by rugk** [comment]
Maybe change the doc, so it is clear "ecmaVersion" does not imply "es6" globals.

So the position in the doc:
![grafik](https://user-images.githubusercontent.com/11966684/39096142-28b86006-464c-11e8-90ad-aed73e469040.png)
https://eslint.org/docs/user-guide/configuring#specifying-environments

Maybe change it to:
> es6 - enable all ECMAScript 6 features except for modules (this automatically sets the ecmaVersion parser option to 6, **but not vice-versa**).

**Comment by reks-scripts** [comment]
I agree with rugk.  The documentation could use some clarification...  I just got hung up on the exact same issue.  I removed "es6": true from my config after reading the same line rugk pointed out assuming that parserOptions.ecmaVersion: 8 would override/include the es6 features and then some.
